From 20a7c7862b1d4c3496d6dd4b7520dd15b87e3a6f Mon Sep 17 00:00:00 2001
From: "Srivatsa S. Bhat (VMware)" <srivatsa@csail.mit.edu>
Date: Fri, 23 Oct 2020 15:12:41 -0700
Subject: [PATCH] apparmor: Fix build failure in apparmor

Apparmor build fails on linux 5.9-rc7 with this error message:

+{"checkpoint_restore", CAP_CHECKPOINT_RESTORE, NO_BACKMAP_CAP, CAPFLAG_BASE_FEATURE},
Error: new capabilities detected please update base_cap_names.h with values from generated_cap_names.h

Fix this by adding CAP_CHECKPOINT_RESTORE to base_cap_names.h as
suggested by the error message above.

Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---
diff -Naurp apparmor-3.0.0/parser/base_cap_names.h apparmor-3.0.0-modified/parser/base_cap_names.h
--- apparmor-3.0.0/parser/base_cap_names.h	2020-10-01 09:50:10.000000000 -0700
+++ apparmor-3.0.0-modified/parser/base_cap_names.h	2020-10-23 15:10:41.919121594 -0700
@@ -8,6 +8,8 @@
 
 {"bpf", CAP_BPF, CAP_SYS_ADMIN, CAPFLAG_BASE_FEATURE},
 
+{"checkpoint_restore", CAP_CHECKPOINT_RESTORE, NO_BACKMAP_CAP, CAPFLAG_BASE_FEATURE},
+
 {"chown", CAP_CHOWN, NO_BACKMAP_CAP, CAPFLAG_BASE_FEATURE},
 
 {"dac_override", CAP_DAC_OVERRIDE, NO_BACKMAP_CAP, CAPFLAG_BASE_FEATURE},

